code/datums/components/rad_vision.dm

```diff
@@ -0,0 +1,38 @@
+/datum/component/rad_vision
+
+/datum/component/rad_vision/Initialize()
+	START_PROCESSING(SSradiation, src)
```

Comment:
`return COMPONENT_INCOMPATIBLE` in initialize if the parent type isnt what you expect